spyder/plugins/updatemanager/widgets/update.py

```diff
@@ -0,0 +1,603 @@
+                f.write(' '.join(sub_cmd))
+            os.chmod(tmpscript, 0o711)  # set executable permissions
+
+            cmd = ['open', '-b', 'com.apple.terminal', tmpscript]
+        else:
+            cmd = ['gnome-terminal', '--window', '--'] + sub_cmd
+
+        subprocess.Popen(' '.join(cmd), shell=True)
+
+
+class UpdateMessageBox(QMessageBox):
+    def __init__(self, icon=None, text=None, parent=None):
+        super().__init__(icon=icon, text=text, parent=parent)
+        self.setWindowModality(Qt.NonModal)
+        self.setTextFormat(Qt.RichText)
+
+
+class UpdateMessageCheckBox(MessageCheckBox):
+    def __init__(self, icon=None, text=None, parent=None):
+        super().__init__(icon=icon, text=text, parent=parent)
+        self.setTextFormat(Qt.RichText)
+        self._parent = parent
+        self.set_checkbox_text(_("Check for updates at startup"))
+        self.option = 'check_updates_on_startup'
+        self.accepted.connect(self.accept)  # ??? Why is the signal necessary?
+        if self._parent is not None:
+            self.set_checked(parent.get_conf(self.option))
+
+    def accept(self):
+        if self._parent is not None:
+            self._parent.set_conf(self.option, self.is_checked())
+
+
+class ProgressDialog(UpdateMessageBox):
+    """Update progress installation dialog."""
+
+    def __init__(self, parent, text):
+        super().__init__(icon=QMessageBox.NoIcon, text=text, parent=parent)
+        self.setWindowTitle(_("Spyder update"))
+        self.setWindowTitle(_("Spyder update"))
+
+        self._progress_bar = QProgressBar(self)
+        self._progress_bar.setMinimumWidth(250)
+        self._progress_bar.setFixedHeight(15)
+
+        layout = self.layout()
+        layout.addWidget(self._progress_bar, 1, 1)
+
+        self.cancel = QPushButton(_("Cancel"))
+        self.okay = QPushButton(_("OK"))
+        self.addButton(self.okay, QMessageBox.YesRole)
+        self.addButton(self.cancel, QMessageBox.NoRole)
+        self.setDefaultButton(self.okay)
+
+        self.show()
+
+    def update_progress(self, progress, total):
+        """Update installation progress bar."""
+        self._progress_bar.setMaximum(total)
+        self._progress_bar.setValue(progress)
+
+
+def error_messagebox(parent, error_msg):
+    box = UpdateMessageBox(
+        icon=QMessageBox.Warning, text=error_msg, parent=parent
+    )
+    self.setWindowTitle(_("Spyder update error"))
+    box.setStandardButtons(QMessageBox.Ok)
+    box.setDefaultButton(QMessageBox.Ok)
+    box.show()
+    return box
+
+
+def info_messagebox(parent, message, version=None, checkbox=False):
+    box_class = UpdateMessageCheckBox if checkbox else UpdateMessageBox
+    message = HEADER.format(version) + message if version else message
+    box = box_class(icon=QMessageBox.Information, text=message, parent=parent)
+    box.setWindowTitle((_("New Spyder version"))
+    box.setWindowTitle((_("New Spyder version"))
```

Comment:
```suggestion
    box.setWindowTitle(_("New Spyder version"))
```

This was probably my mistake during review. That's why tests are failing.

Reply:
I caught it, but I'm not done with changes yet anyway. Thanks.